src/lib/y2partitioner/actions/delete_bcache.rb

```diff
@@ -21,48 +21,66 @@
 
 require "yast"
 require "y2partitioner/actions/delete_device"
-require "y2partitioner/device_graphs"
+require "y2partitioner/actions/controllers/bcache"
 
 module Y2Partitioner
   module Actions
-    # Action for deleting a Bcache
-    #
-    # @see DeleteDevice
+    # Action for deleting a Bcache, see {Actions::DeleteBcache}
     class DeleteBcache < DeleteDevice
-      def initialize(*args)
+      # Constructor
+      #
+      # @param bcache [Y2Storage::Bcache]
+      def initialize(bcache)
         super
+
         textdomain "storage"
+
+        @bcache_controller = Controllers::Bcache.new(bcache)
       end
 
     private
 
-      # @see DeleteDevice#errors
+      # @return [Controllers::Bcache]
+      attr_reader :bcache_controller
+
+      # @see Actions::Base#errors
       def errors
-        errors = super + [shared_cset]
+        (super + [flash_only_error, unsafe_detach_error]).compact
+      end
+
+      # Error when the bcache is Flash-only
+      #
+      # @return [String, nil] nil if the bcache is not Flash-only.
+      def flash_only_error
+        return nil unless flash_only_bcache?
 
-        errors.compact
+        # TRANSLATORS: error message, %{name} is replaced by a bcache name (e.g., /dev/bcache0)
+        format(
+          _("The device %{name} is a Flash-only Bcache. Deleting this kind of devices\n" \
+            "is not supported yet."),
+          name: bcache_controller.bcache.name
+        )
       end
 
-      # Error when the bcache shares cset with any other bcache.
+      # Error when the caching set cannot be detached safely
       #
       # @see doc/bcache.md
-      # @return [String, nil] nil if there is no error
-      def shared_cset
-        return nil unless device.bcache_cset
-        system = Y2Partitioner::DeviceGraphs.instance.system
-        return nil if single_bcache_cset? || !device.exists_in_devicegraph?(system)
+      #
+      # @return [String, nil] nil if detach is safe
+      def unsafe_detach_error
+        return nil if safe_detach_bcache?
 
-        # TRANSLATORS: Error when trying to modify an existing bcache that shares caches.
+        # TRANSLATORS: Error message when detach is not a safe action
         _(
-          "The bcache shares its cache set with other devices.\nThis can result in unreachable space " \
-            "if done without detaching.\nDetaching can take a very long time in some situations."
+          "The bcache cannot be deleted because it shares its cache set with other devices.\n" \
+          "Detaching is required to avoid possible unreachable space, but detaching action\n" \
+          "could take a very long time in some situations."
```

Comment:
The sentence, specially the part  "but detaching action could take..." sounds a little bit strange to me.

I would merge the old and the new sentence in something like this.
```
The bcache cannot be deleted because it shares its cache set with other devices.\n
Deleting it without detaching the device first can result in unreachable space.\n
Unfortunately detaching can take a very long time in some situations.
```